x-pack/test/scalability/report_parser.ts

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ToolingLog } from '@kbn/tooling-log';
+import fs from 'fs';
+import { ScalabilitySetup, ResponseTimeMetric } from '@kbn/journeys';
+import { CapacityMetrics, DataPoint, ResponseMetric, RpsMetric } from './types';
+
+const RESPONSE_METRICS_NAMES = [
+  'min',
+  '25%',
+  '50%',
+  '75%',
+  '80%',
+  '85%',
+  '90%',
+  '95%',
+  '99%',
+  'max',
+];
+const DEFAULT_THRESHOLD = {
+  threshold1: 3000,
+  threshold2: 9000,
+  threshold3: 15000,
+};
+const DEFAULT_METRIC = '85%';
+const REQUESTS_REGEXP = /(?<=var requests = unpack\(\[)(.*)(?=\]\);)/g;
+const RESPONSES_PERCENTILES_REGEXP =
+  /(?<=var responsetimepercentilesovertimeokPercentiles = unpack\(\[)(.*)(?=\]\);)/g;
+
+const getRPSByResponseTime = (
+  rpsData: RpsMetric[],
+  responseTimeData: ResponseMetric[],
+  responseTimeThreshold: number,
+  metricName: ResponseTimeMetric,
+  defaultRpsValue: number
+) => {
+  const timestamp = getTimePoint(responseTimeData, metricName, responseTimeThreshold);
+  if (timestamp === -1) {
+    // Data point was not found, probably 'responseTimeThreshold' is too high or maxRps is too low for the api
```

Comment:
I wonder if it couldn't be misleading if you look at a dashboard and see the `maxRps` for what you think is a specific `responseTimeThreshold`. Especially because the test config never sets an explicit maxRps target it might be hard to know if your tests should use more concurrent users or lower the thresholds.

I'm not entirely sure what's the best behaviour here. But maybe returning `-1` like below could be more useful since someone would definitely investigate if they saw it on their dashboard? Might also be useful with a log entry in the CI build.